.ci/lint-python.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/sh
+
+DIR_TO_CHECK=${1}
+
+echo "running flake8"
+flake8 \
+    --ignore=E501,W503 \
+    --exclude=./.nuget,./external_libs,./python-package/build,./python-package/compile \
+    "${DIR_TO_CHECK}" \
+|| exit -1
+echo "done running flake8"
+
+echo "running pydocstyle"
+pydocstyle \
+    --convention=numpy \
+    --add-ignore=D105 \
+    --match-dir="^(?!^external_libs|test|example).*" \
+    --match="(?!^test_|setup).*\.py" \
+    "${DIR_TO_CHECK}" \
+|| exit -1
+echo "done running pydocstyle"
+
+echo "running isort"
+isort \
+    --check-only \
+    "${DIR_TO_CHECK}" \
+|| exit -1
+echo "done running isort"
+
+echo "running mypy"
+mypy \
+    --ignore-missing-imports \
+    --exclude 'build/' \
+    --exclude 'compile/' \
+    --exclude 'docs/' \
+    --exclude 'examples/' \
+    --exclude 'external_libs/' \
+    --exclude 'tests/' \
+    "${DIR_TO_CHECK}" \
```

Comment:
Wouldn't it be better to leave it with just `python-package` as it is now? I sometimes have other files at the repo root but I run mypy only on the python-package folder, so it'd be nice to have it check only `"${DIR_TO_CHECK}/python-package"` instead.

Reply:
```suggestion
    --exclude 'build/' \
    --exclude 'compile/' \
    --exclude 'docs/' \
    --exclude 'examples/' \
    --exclude 'external_libs/' \
    --exclude 'tests/' \
    "${DIR_TO_CHECK}/python-package" \
```

Ok yeah I'm good with that. That does mean that `DIR_TO_CHECK` now cannot be "any random directory of Python code", but I think that's totally fine for this script with a very specific purpose.